docs/source/debugging.md

```diff
@@ -0,0 +1,183 @@
+* [launch.json in hi-ml-azure](https://github.com/microsoft/hi-ml/tree/main/hi-ml-azure/.vscode/lanch.json)
+* [launch.json in hi-ml-histopathology](https://github.com/microsoft/hi-ml/tree/main/hi-ml-histopathology/.vscode/lanch.json)
+
+VS Code restricts debugging to user-written code only by default. If you want to step through external code and
+standard libraries functions, set `"justMyCode": false` inside the debugging config block in the `launch.json` file.
+
+In particular, if you would like to debug the current file while breaking through `pytorch` code, navigate to
+`himl-projects.code-workspace` in the repo root and edit the "launch" block as follow:
+
+```json
+"launch": {
+    "configurations": [
+      {
+        "name": "Python: Current File",
+        "type": "python",
+        "request": "launch",
+        "program": "${file}",
+        "console": "integratedTerminal",
+        "justMyCode": false
+      },
+    ],
+}
+```
+
+### Pytorch Lightning flags for debugging and quick runs
+
+The hi-ml toolbox is built upon [Pytorch Lightning (PL)](https://www.pytorchlightning.ai/) to help you build scalable
+deep learning models for healtcare and life sciences. Refer to [Running ML Experiments with hi-ml](runner.md) for
+detailed instructions on how to build scalable pipelines within hi-ml.
+
+Whether you're building a brand new model, or extending an existing one, you might want to make sure that your
+code runs as expected locally before submitting a job to AzureML. hi-ml supports a set of debugging flags that triggers
+[Pytorch Lightning Trainer](https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#trainer) arguments to
+help you detect any potential errors or bugs at early stage.
+
+These are available as part of the
+[TrainerParams](https://github.com/microsoft/hi-ml/blob/96b8ba60ebf84416f5c9b13e2df15ee07a13e6bb/hi-ml/src/health_ml/deep_learning_config.py#L357)
+and can be used as extra command line arguments with the [hi-ml
+runner](https://github.com/microsoft/hi-ml/blob/746c8b58c1af71f71eeaaac2a8584be1d9a5386f/hi-ml/src/health_ml/runner.py#L107).
+
+* `pl-fast-dev-run`: If set to `n`, runs the pipeline for only `n` batch(es) of train, val and test for only a single
+  epoch. Additionally [this flag](https://pytorch-lightning.readthedocs.io/en/stable/common/debugging.html#fast-dev-run)
+  disables all callbacks and hyperparameters serialization which makes the debugging process very quick. This must be
+  used for debugging purposes only.
+* `pl-limit-train-batches`: Limits the training dataset to the given number of batches `n`.
+* `pl-limit-val-batches`: Limits the validation dataset to the given number of batches `n`.
+* `pl-limit-train-batches`: Limits the test dataset to the given number of batches `n`.
+
+In general, it is very useful to run the following two steps as part of the developement cycle:
+
+1. Make sure all training, validation and test loops complete properly:
```

Comment:
I dont' think we should add it to the Makefile because these are "toy" commands (YourcustomContainer doesn't exist for instance) to show how to debug using the hi-ml runner and the flags in TrainerParams.  

I'm afraid specifying only the flags might be confusing. which script to launch? 

What if we change HimlHisto by YourEnvName?

Reply:
We should definitely remove HimlHisto (to be it might be overkill to tell the user to activate their environment anyway?) but my concern is that it's not clear which are the standard runner args and which are debugging args